o365-auth/src/main/java/com/microsoft/office365/auth/AuthenticationManager.java

```diff
@@ -4,146 +4,182 @@
  */
 package com.microsoft.office365.auth;
 
+import android.accounts.AuthenticatorException;
+import android.accounts.OperationCanceledException;
 import android.app.Activity;
-import android.content.Context;
-import android.content.SharedPreferences;
+import android.util.Log;
 
-import com.microsoft.aad.adal.ADALError;
-import com.microsoft.aad.adal.AuthenticationCallback;
-import com.microsoft.aad.adal.AuthenticationContext;
-import com.microsoft.aad.adal.AuthenticationException;
-import com.microsoft.aad.adal.AuthenticationResult;
-import com.microsoft.aad.adal.PromptBehavior;
+import com.microsoft.graph.authentication.IAuthenticationProvider;
+import com.microsoft.graph.http.IHttpRequest;
+import com.microsoft.identity.client.AuthenticationCallback;
+import com.microsoft.identity.client.AuthenticationResult;
+import com.microsoft.identity.client.IAccount;
+import com.microsoft.identity.client.PublicClientApplication;
+import com.microsoft.identity.client.exception.MsalException;
 
-import static com.microsoft.aad.adal.AuthenticationResult.AuthenticationStatus.Succeeded;
+import java.io.IOException;
 
-public class AuthenticationManager {
+public class AuthenticationManager implements IAuthenticationProvider{
 
-    private static final String USER_ID_VAR_NAME = "userId";
+    private final Activity mActivity;
 
-    private static final int PREFERENCES_MODE = Context.MODE_PRIVATE;
+    private final PublicClientApplication mPublicClientApplication;
 
-    private final Activity mActivity;
+    private final String mClientId;
```

Comment:
We can also remove `mClientId` as it is only used for in the construction of  `mPublicClientApplication`

Reply:
Done! Thanks